src/ivc/public_params.rs

```diff
@@ -1,71 +1,59 @@
-use std::{fmt, num::NonZeroUsize};
+use std::{fmt, marker::PhantomData, num::NonZeroUsize};
 
-use ff::{Field, FromUniformBytes, PrimeFieldBits};
+use ff::{FromUniformBytes, PrimeFieldBits};
 use group::prime::PrimeCurveAffine;
-use halo2_proofs::plonk::{Column, Instance};
 use halo2curves::CurveAffine;
 use log::*;
 use serde::Serialize;
 
 use crate::{
     commitment::CommitmentKey,
     digest::{self, DigestToCurve},
-    ivc::step_circuit::StepCircuitExt,
+    main_gate::MainGateConfig,
     plonk::PlonkStructure,
     poseidon::ROPair,
     table::TableData,
 };
 
-use super::{
-    step_circuit::{self, StepConfig, SynthesizeStepParams},
-    StepCircuit,
-};
+use super::{step_folding_circuit::StepParams, StepCircuit};
 
 #[derive(Debug, thiserror::Error)]
-pub enum Error {
-    #[error("While configure table data: {0:?}")]
-    Configure(#[from] step_circuit::ConfigureError),
-}
+pub enum Error {}
 
-pub(crate) struct CircuitPublicParams<'key, const ARITY: usize, const MAIN_GATE_T: usize, C, SC, RP>
+pub(crate) struct CircuitPublicParams<'key, const ARITY: usize, const MAIN_GATE_T: usize, C, RP>
 where
     C: CurveAffine,
     C::Scalar: PrimeFieldBits + FromUniformBytes<64>,
-    SC: StepCircuit<ARITY, C::Scalar>,
     RP: ROPair<C::Scalar>,
 {
+    pub k_table_size: u32,
     pub ck: &'key CommitmentKey<C>,
-    pub params: SynthesizeStepParams<C::Scalar, RP::OnCircuit>,
-
-    td: TableData<C::Scalar>,
-    X0: Column<Instance>,
-    config: StepConfig<ARITY, C::Scalar, SC, MAIN_GATE_T>,
+    pub params: StepParams<C::Scalar, RP::OnCircuit>,
```

Comment:
We can move `ro_constant` from `StepParam` into `CircuitPublicParams. It will still be hashed into PublicParam hash.

Reply:
We need `ro_constant` at each step to create on-circuit oracles. `StepParams` is just that part of `CircuitPublicParams` which should be passed at each step. And it is passed at each step